patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -837,13 +837,18 @@
 +     */
 +    public boolean canPlaceTorchOnTop(IBlockState state, IBlockAccess world, BlockPos pos)
 +    {
-+        if (state.func_185896_q() || state.func_193401_d(world, pos, EnumFacing.UP) == BlockFaceShape.SOLID)
++        if (this == Blocks.field_185775_db || this == Blocks.field_150428_aP)
 +        {
-+            return this != Blocks.field_185775_db && this != Blocks.field_150428_aP;
++            return false;
++        }
++        else if (state.func_185896_q() || this instanceof BlockFence || this == Blocks.field_150359_w || this == Blocks.field_150463_bK || this == Blocks.field_150399_cn)
++        {
++            return true;
 +        }
 +        else
 +        {
-+            return this instanceof BlockFence || this == Blocks.field_150359_w || this == Blocks.field_150463_bK || this == Blocks.field_150399_cn;
++            BlockFaceShape shape = state.func_193401_d(world, pos, EnumFacing.UP);
++            return (shape == BlockFaceShape.SOLID || shape == BlockFaceShape.CENTER || shape == BlockFaceShape.CENTER_BIG) && !func_193384_b(this);
```

Comment:
These two lines will never be executed as well

Reply:
Same here, the `return this instanceof...` is being removed.